File: api/src/main/java/com/bindglam/neko/api/config/ComponentConfigurable.java
package com.bindglam.neko.api.config;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.jetbrains.annotations.Nullable;

public class ComponentConfigurable implements Configurable<Component, String> {
    @Override
    public @Nullable Component load(@Nullable String value) {
        if(value == null) return null;

        return MiniMessage.miniMessage().deserialize(value);
    }
}

File: core/src/main/kotlin/com/bindglam/neko/config/ComponentConfigurable.kt
package com.bindglam.neko.config

import com.bindglam.neko.api.config.Configurable
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage

class ComponentConfigurable : Configurable<Component, String> {
    override fun load(value: String?): Component? = value?.let { MiniMessage.miniMessage().deserialize(it) }
}